pilot/pkg/model/authentication.go

```diff
@@ -39,6 +52,61 @@ func GetConsolidateAuthenticationPolicy(store IstioConfigStore, service *Service
 	return nil
 }
 
+// ConstructSdsSecretConfig constructs SDS Sececret Configuration.
+func ConstructSdsSecretConfig(serviceAccount string, refreshDuration *time.Duration, sdsUdsPath string) *auth.SdsSecretConfig {
+	if serviceAccount == "" || sdsUdsPath == "" || refreshDuration == nil {
+		return nil
+	}
+
+	return &auth.SdsSecretConfig{
+		Name: serviceAccount,
+		SdsConfig: &core.ConfigSource{
+			ConfigSourceSpecifier: &core.ConfigSource_ApiConfigSource{
+				ApiConfigSource: &core.ApiConfigSource{
+					ApiType: core.ApiConfigSource_GRPC,
+					GrpcServices: []*core.GrpcService{
+						{
+							TargetSpecifier: &core.GrpcService_GoogleGrpc_{
+								GoogleGrpc: &core.GrpcService_GoogleGrpc{
+									TargetUri:  sdsUdsPath,
+									StatPrefix: SDSStatPrefix,
+									CallCredentials: []*core.GrpcService_GoogleGrpc_CallCredentials{
+										&core.GrpcService_GoogleGrpc_CallCredentials{
+											CredentialSpecifier: &core.GrpcService_GoogleGrpc_CallCredentials_GoogleComputeEngine{
+												GoogleComputeEngine: &types.Empty{},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+					RefreshDelay: refreshDuration,
+				},
+			},
+		},
+	}
+}
+
+// ConstructValidationContext constructs ValidationContext in CommonTlsContext.
+func ConstructValidationContext(rootCAFilePath string, subjectAltNames []string) *auth.CommonTlsContext_ValidationContext {
+	ret := &auth.CommonTlsContext_ValidationContext{
+		ValidationContext: &auth.CertificateValidationContext{
+			TrustedCa: &core.DataSource{
+				Specifier: &core.DataSource_Filename{
+					Filename: rootCAFilePath,
+				},
+			},
+		},
+	}
+
+	if len(subjectAltNames) > 0 {
```

Comment:
And I think we can set VerifySubjectAltName even if subjectAltNames is empty, right? Then we don't even need this if clause. Another good point to have unit test :)

Reply:
ditto.